### PR TITLE
[SPARK-31566][SQL][DOCS] Add SQL Rest API Documentation

### PR DESCRIPTION
**Pull request by erenavsarogullari**
### What changes were proposed in this pull request?
SQL Rest API exposes query execution details and metrics as Public API. Its documentation will be useful for the end-users.

### Why are the changes needed?
SQL Rest API does not exist under Spark Rest API.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Manually build and check

### CODE REVIEW

**Comment by erenavsarogullari** [comment]
cc @gengliangwang @vanzin @dongjoon-hyun

**Comment by HyukjinKwon** [comment]
I think you can just fold this into #28208.

**Comment by erenavsarogullari** [comment]
Hi @HyukjinKwon,
Thanks for the review. I removed the dependency so current documentation just points the existing functionality on `SqlResource`. So, this can go independently now.

**Comment by HyukjinKwon** [comment]
cc @gengliangwang FYI

**Comment by gengliangwang** [comment]
I think either a separate PR or folding into #28208 is fine. 
Since #28208 is not merged yet, let's mark this one as WIP.

**Comment by gengliangwang** [comment]
@erenavsarogullari You can continue on this one now
Please remove the `[WIP]` in the title and provide a screenshot of doc after built

**Comment by erenavsarogullari** [comment]
Hi @gengliangwang,

This PR is up-to-date. Please find doc screenshot as attached
Doc Link: https://github.com/erenavsarogullari/spark/blob/SPARK-31566/docs/monitoring.md
Spark SQL Rest API Doc

File: docs/monitoring.md [hunk 1]
```diff
@@ -544,6 +544,24 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
     <td><code>/applications/[app-id]/streaming/batches/[batch-id]/operations/[outputOp-id]</code></td>
     <td>Details of the given operation and given batch.</td>
   </tr>
+  <tr>
+    <td><code>/applications/[app-id]/sql</code></td>
+    <td>A list of all queries for a given application.
+    <br>
+    <code>?details=[true (default) | false]</code> lists metric details in addition to queries details.
```
Comment (gengliangwang) [inline docs/monitoring.md:551]: how about changing 
```
lists metric details in addition to queries details
```
to
```
lists/hides details of Spark plan nodes.
```

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**